pkg/controller/servicemesh/webhooks/validation/memberroll.go

```diff
@@ -107,32 +117,103 @@ func (v *MemberRollValidator) Handle(ctx context.Context, req atypes.Request) at
 	}
 	if !allowed {
 		// check each namespace separately, but only check newly added namespaces
-		namespacesToCheck := sets.NewString(smmr.Spec.Members...)
-
-		if req.AdmissionRequest.Operation == admissionv1.Update {
-			oldSmmr := &maistrav1.ServiceMeshMemberRoll{}
-			err := v.decoder.DecodeRaw(req.AdmissionRequest.OldObject, oldSmmr)
-			if err != nil {
-				logger.Error(err, "error decoding old object in admission request")
-				return admission.ErrorResponse(http.StatusBadRequest, err)
-			}
-			namespacesToCheck.Delete(oldSmmr.Spec.Members...)
+		namespacesToCheck, err := v.findNewlyAddedNamespaces(smmr, req)
+		if err != nil {
+			return admission.ErrorResponse(http.StatusInternalServerError, err)
```

Comment:
Should we log the error in this case? Also, returning a UnprocessableEntity error might give more context than internal server error, but I'm not sure what the standard behaviour is for k8s webhooks

Reply:
You're right. This should be a 400 BadRequest, since the error occurs only when the oldObject can't be decoded. 

But because this is a client error, it shouldn't be logged as an error. The user gets the error message in the response and the cluster-admin that manages the operator doesn't want to see client errors in the log. Right?

Reply:
Ah, right. No logging then!

Reply:
Good question, but this is an internal error, not a client error.  Client isn't sending oldObject, the apiserver is.  That said, there's not much we can do about it either, so does it need to be logged, and as you said, the user gets the the error?